staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -354,7 +355,9 @@ func (a *Authenticator) refreshCredsLocked(r *clientauthentication.Response) err
 	// Only close all connections when TLS cert rotates. Token rotation doesn't
 	// need the extra noise.
 	if a.onRotate != nil && oldCreds != nil && !reflect.DeepEqual(oldCreds.cert, a.cachedCreds.cert) {
+		metrics.ClientCertRotationAge.Observe(oldCreds.cert.Leaf.NotAfter.Sub(oldCreds.cert.Leaf.NotBefore))
```

Comment:
the prometheus metric wired to this says "Count of the number of seconds the last client certificate lived before being rotated.", but this looks like it is reporting the valid age of the cert. To match the prometheus docs, I expected something like `time.Now().Sub(oldCreds.cert.Leaf.NotBefore)`

Reply:
we probably still want to report metrics even if onRotate is nil, so move that condition to an inner if block wrapping the onRotate call

Reply:
also, oldCreds.cert can be nil if the exec auth plugin only returned token auth

Reply:
Ah, yes great catches again. 

Fixed in next commit.